### DISCUSSION TITLE
Type DBase for OpenSnitch

### DISCUSSION THREAD

**Discussion by Doroshenko-T** [evilsocket/opensnitch#886] [General]
Hello,

Nowhere in the description OpenSnitch is there a description of which DataBase file should be used to store the rules.
I use version  1.6.0rc5.

**Comment by gustavo-iniguez-goya** [comment]
Hi @T-Shilov ,

The rules are stored on disk, as json files, under /etc/opensnitchd/rules/.

They're also saved on the GUI (in sqlite format), but right now only to edit and send them to the daemon.

**Comment by Doroshenko-T** [comment]
Well, thank you! 
But why is OpenSnitch when switching the rules storage in memory to the database, it asks me which file to choose?
I do not know how to answer :-(

**Reply by gustavo-iniguez-goya** [reply]
aah ok, that option is not only for rules, but for all the events, including daemon's rules.

It asks you to choose a file name for the DB that will be used to store on the disk the events and rules.

**Comment by Doroshenko-T** [comment]
But there are a lot of JSON files, and I do not know which one to choose...

**Reply by gustavo-iniguez-goya** [reply]
Sorry @T-Shilov , maybe I'm not explaining it right.

> Nowhere in the description OpenSnitch is there a description of which DataBase file should be used to store the rules.

Rules are only stored under /etc/opensnitchd/rules/ as .json files. From the GUI you only edit or create them, but the content is stored under that directory.

On the other hand, the GUI stores internally on a DB all the data received by the daemon: events, connections, configuration and rules.
But only as a way to display it graphically to the user. This DB is stored on memory (default), or on disk (file).

If you want to store connections for weeks or months, chose db type -> file, and select a directory and a name for the DB (for example navigate to /home/YOUR_USER/ and type -> opensnitch.db).
